src/audio/eq_fir/eq_fir.c

```diff
@@ -171,36 +171,27 @@ static inline int set_fir_func(struct processing_module *mod, enum sof_ipc_frame
 static int eq_fir_params(struct processing_module *mod)
 {
 	struct sof_ipc_stream_params *params = mod->stream_params;
-	struct sof_ipc_stream_params comp_params;
 	struct comp_dev *dev = mod->dev;
-	struct comp_buffer *sinkb;
-	struct comp_buffer __sparse_cache *sink_c;
-	enum sof_ipc_frame valid_fmt, frame_fmt;
-	int i, ret;
+	struct comp_buffer *sinkb, *sourceb;
+	struct comp_buffer __sparse_cache *sink_c, *source_c;
 
 	comp_dbg(dev, "eq_fir_params()");
 
-	comp_params = *params;
-	comp_params.channels = mod->priv.cfg.base_cfg.audio_fmt.channels_count;
-	comp_params.rate = mod->priv.cfg.base_cfg.audio_fmt.sampling_frequency;
-	comp_params.buffer_fmt = mod->priv.cfg.base_cfg.audio_fmt.interleaving_style;
+	ipc4_base_module_cfg_to_stream_params(&mod->priv.cfg.base_cfg, params);
 
-	audio_stream_fmt_conversion(mod->priv.cfg.base_cfg.audio_fmt.depth,
-				    mod->priv.cfg.base_cfg.audio_fmt.valid_bit_depth,
-				    &frame_fmt, &valid_fmt,
-				    mod->priv.cfg.base_cfg.audio_fmt.s_type);
+	component_set_nearest_period_frames(dev, params->rate);
 
-	comp_params.frame_fmt = valid_fmt;
-
-	for (i = 0; i < SOF_IPC_MAX_CHANNELS; i++)
-		comp_params.chmap[i] = (mod->priv.cfg.base_cfg.audio_fmt.ch_map >> i * 4) & 0xf;
-
-	component_set_nearest_period_frames(dev, comp_params.rate);
 	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
 	sink_c = buffer_acquire(sinkb);
-	ret = buffer_set_params(sink_c, &comp_params, true);
+	ipc4_update_buffer_format(sink_c, &mod->priv.cfg.base_cfg.audio_fmt);
 	buffer_release(sink_c);
-	return ret;
+
+	sourceb = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
+	source_c = buffer_acquire(sourceb);
+	ipc4_update_buffer_format(source_c, &mod->priv.cfg.base_cfg.audio_fmt);
+	buffer_release(source_c);
+
+	return 0;
```

Comment:
I'm losing track of what the module params should do, but I guess addition of the source fmt update is needed here. This is starting to go outside the original topic of this PR.

Reply:
do you mean lose update fmt? it does update everything with ipc4_update_buffer_format and ipc4_update_buffer_format

Reply:
I mean setting format for "bsource_list". Original code only sets format for bsink_list, but your new code bsource_list and bsink_list. This may be the right thing to do, but a bit confusing to have this in a commit "to simplify parameters".